### PR TITLE
Dispatch Series comparison ops to DatetimeIndex and TimedeltaIndex

### PR DESCRIPTION
**Pull request by jbrockmendel**
Another couple steps towards unifying the implementations.

We will be able to simplify _comp_method_SERIES a lot if/when the Categorical vs CategoricalIndex inconsistencies are sorted out #19513.

### CODE REVIEW

File: pandas/core/ops.py [hunk 1]
```diff
@@ -788,17 +782,35 @@ def wrapper(self, other, axis=None):
...
+            return _construct_result(self, res_values,
+                                     index=self.index, name=res_name,
+                                     dtype=res_values.dtype)
+
         elif isinstance(other, ABCSeries):
-            name = com._maybe_match_name(self, other)
-            if not self._indexed_same(other):
-                msg = 'Can only compare identically-labeled Series objects'
-                raise ValueError(msg)
+            # By this point we know that self._indexed_same(other)
```
Comment (gfyoung) [inline pandas/core/ops.py]: Comma between "point" and "we"

File: pandas/core/ops.py [hunk 2]
```diff
@@ -788,17 +782,35 @@ def wrapper(self, other, axis=None):
...
 
+        res_name = _get_series_op_result_name(self, other)
+
         if isinstance(other, ABCDataFrame):  # pragma: no cover
             # Defer to DataFrame implementation; fail early
             return NotImplemented
 
+        elif isinstance(other, ABCSeries) and not self._indexed_same(other):
+            raise ValueError('Can only compare identically-labeled Series '
+                             'objects')
```
Comment (gfyoung) [inline pandas/core/ops.py]: Nit: I prefer not to "orphan" words on their own line.  Move the "Series" part of your sentence there too.

File: pandas/tests/indexes/datetimes/test_partial_slicing.py [hunk 3]
```diff
@@ -349,7 +349,7 @@ def test_loc_datetime_length_one(self):
 
     @pytest.mark.parametrize('datetimelike', [
         Timestamp('20130101'), datetime(2013, 1, 1),
-        date(2013, 1, 1), np.datetime64('2013-01-01T00:00', 'ns')])
+        np.datetime64('2013-01-01T00:00', 'ns')])
```
Comment (gfyoung) [inline pandas/tests/indexes/datetimes/test_partial_slicing.py]: Why did you have to change the parametrization?

File: pandas/core/ops.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/ops.py]: Will change.  Ditto comma above.

File: pandas/tests/indexes/datetimes/test_partial_slicing.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/datetimes/test_partial_slicing.py]: b/c ATM `Series[datetime64].__cmp__(date)` treats the `date` as a `datetime`, i.e. allows the comparison.  But `DatetimeIndex` does not -- following convention set by `Timestamp` (and `datetime` itself).  The `DatetimeIndex` behavior is canonical.

**Comment by jbrockmendel** [comment]
I can't seem to replicate the test failure.  Can anyone else?

```
ser = pd.Series([pd.Timestamp('2000-01-29 01:59:00'), 'NaT'])
result = ser != ser
assert (result == [False, True]).all()
```

**Comment by jbrockmendel** [comment]
I'm having no luck with replicating the error.  Closing for now, will revisit when the queue dies down.

**Comment by jbrockmendel** [comment]
Most recent push was intended for my circleCI account, forgot it would get pushed here.  Will revert.  No luck SSHijg into circleCI so far.

**Pull request closed without merging**